### ISSUE TITLE
`:stopinsert` does not clean up insert mode properly.

### ISSUE THREAD

**Issue by tjdevries** [neovim/neovim#12976]
### Steps to reproduce using `nvim -u NORC`

```

" hello
" helper
" howdy
" hunting

call timer_start(2000, { -> execute("stopinsert") })

" On this line, after sourcing this file, enter ins-completion mode within 2 seconds.
" After entering, you will not be able to type ever again in vim
" h
```
## Actual Behavior

You can no longer type

## Expected Behavior

Ability to type in my text editor.

**Comment by Happy-Dude** [comment]
Heyo, we got a patch from upstream: https://github.com/vim/vim/commit/d0e1b7103c14eb0d175c6b245b4b6ed93a204da9

**Comment by Happy-Dude** [comment]
A follow-up patch from upstream: https://github.com/vim/vim/commit/4537bcc88956f86267c25edf8008e0dbde598652 that addresses vim/vim#7035

**Comment by tjdevries** [comment]
@Happy-Dude Do you know where we ended up on this? I am lost in the trail of PRs :laughing:

**Comment by Happy-Dude** [comment]
@tjdevries carrying what we talked about in stream: I tried to port the relevant patches into https://github.com/neovim/neovim/pull/13027 -- but it was more involved than I expected.

A contributor with a better understanding of the vim `prompt buffer` feature probably needs to take a look at it and understand/figure out how they interact with neovim's floating windows. When I ported the patches as it, some of the neovim window tests started failing.
